### PR TITLE
AS-327: PFB import integration tests [risk: no]

### PR DESCRIPTION
**Pull request by davidangb**
Well it took 24 commits, but all tests are passing now.

```
[info] PFBImportSpec:
[info] Orchestration
[info]   should import a PFB file via import service
[info]   - for the owner of a workspace (1 minute, 34 seconds)
[info]   - for writers of a workspace (1 minute, 20 seconds)
[info]   should asynchronously result in an error when attempting to import a PFB file via import service
[info]   - if the file to be imported is invalid (12 seconds, 773 milliseconds)
[info]   should synchronously return an error when attempting to import a PFB file via import service
[info]   - for readers of a workspace (12 seconds, 11 milliseconds)
[info]   - with an invalid POST payload (11 seconds, 91 milliseconds)
```

also, what are you talking about, Coveralls? Coverage decreased? All this PR does is add tests. Grr.

---

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by davidangb** [comment]
jenkins retest

**Comment by davidangb** [comment]
jenkins retest

**Comment by davidangb** [comment]
Jenkins retest

**Comment by davidangb** [comment]
jenkins retest

**Comment by davidangb** [comment]
now that broadinstitute/firecloud-develop#2273 is merged, jenkins retest

**Comment by davidangb** [comment]
modified some IAM permissions in the import service project; jenkins retest

**Comment by ichengchang** [comment]
LGTM

**Review by ichengchang — APPROVED** [review]
LGTM

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PFBImportSpec.scala [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            val importJobId: String = importJobIdValues.head match {
+              case js:JsString => js.value
+              case x => fail("got in invalid jobId: " + x.toString())
+            }
+
+            // poll for completion as owner
+            eventually {
+              val resp: HttpResponse = Orchestration.getRequest( s"${workspaceUrl(projectName, workspaceName)}/importPFB/$importJobId")
+              resp.status shouldBe StatusCodes.OK
+              blockForStringBody(resp).parseJson.asJsObject.fields.get("status").value shouldBe JsString("Error")
```
Comment (ungwudik) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PFBImportSpec.scala:127]: Nit: Is there a reason the error message here isn't more descriptive, like the other negative tests?

**Review by ungwudik — APPROVED** [review]

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PFBImportSpec.scala (hunk 1 above)
Comment (davidangb) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PFBImportSpec.scala:127]: The negative tests are looking at the text message contained in an `ErrorReport` object, which as you notice can be descriptive and human-readable. This test is looking only at the `status` value for the import job, which is an enum and constrained to specific values like `Done`, `Error`, and `Upserting`. I thought this test should only check that attempting to import a nonexistent file results in an error ... if Import Service changes its wording on the error message description, it won't break the test.

**Pull request merged**